### PR TITLE
circulation: add ill requests tab

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds a new tab to display pending ILL requests.
* Fixes the problem on separator parameter in JournalVolume pipe.
* Fixes the typing of the journal parameter to the JournalVolume pipe.
* Closes rero/rero-ils#2165.

### Dependencies
ILS: https://github.com/rero/rero-ils/pull/3322

### CODE REVIEW

File: projects/admin/src/app/api/ill-request-api.service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  /**
+   * Constructor
+   * @param _recordService - RecordService
+   */
+  constructor(private _recordService: RecordService) {
+    super();
+  }
+
+  getByPatronPid(patronPid: string): Observable<Record | Error> {
```
Comment (zannkukai) [inline projects/admin/src/app/api/ill-request-api.service.ts:40]: Return statement seems incorrect. It should return `Observable<Array<Record> | Error>`

File: projects/admin/src/app/circulation/circulation-routing.module.ts [hunk 2]
```diff
@@ -58,6 +59,10 @@ const routes: Routes = [
         path: 'pending',
         component: PendingComponent,
         canActivate: [ PermissionGuard ], data: { permissions: [ PERMISSIONS.CIRC_ADMIN ] }
+      },{
+        path: 'ill',
+        component: IllRequestComponent,
+        canActivate: [ PermissionGuard ], data: { permissions: [ PERMISSIONS.CIRC_ADMIN ] }
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/circulation-routing.module.ts:65]: Maybe add the permissions "ILL_SEARCH" ?

File: projects/admin/src/app/circulation/patron/ill-request/ill-request-item/ill-request-item.component.html [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<div class="row item">
+  <div class="col-sm-6" translate>
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/ill-request/ill-request-item/ill-request-item.component.html:18]: In UI, we could skip "sm", "md", "lg", ... --> "col-6" seems suffisent

File: projects/admin/src/app/circulation/patron/ill-request/ill-request-item/ill-request-item.component.html [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    >
+      <i class="fa fa-eye" aria-hidden="true"></i>
+    </a>
+  </div>
+  <div name="collapsed-details" class="col-12 mt-2" *ngIf="!isCollapsed">
+    <dl class="row">
+      <!-- PUBLISHER -->
+      <ng-container *ngIf="record.metadata.document.publisher">
+        <dt class="offset-1 col-sm-3 col-md-2 col-lg-1 label-title" translate>Publisher</dt>
+        <dd class="col-sm-8 col-md-9 col-lg-10 mb-0">{{ record.metadata.document.publisher }}</dd>
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/ill-request/ill-request-item/ill-request-item.component.html:40]: remove sm, md, lg --> just use a simple "col-x"

**Review by zannkukai — CHANGES_REQUESTED** [review]

File: projects/admin/src/app/api/ill-request-api.service.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/admin/src/app/api/ill-request-api.service.ts:40]: The signature of the `getRecords` function is `Observable<Record | Error>`.
Ref: https://github.com/rero/ng-core/blob/staging/projects/rero/ng-core/src/lib/record/record.service.ts#L116

File: projects/admin/src/app/circulation/circulation-routing.module.ts (hunk 2 above)
Comment (Garfield-fr) [reply projects/admin/src/app/circulation/circulation-routing.module.ts:65]: All routes have the `PERMISSIONS.CIRC_ADMIN` permission. Is it different for the ILL route ?

**Review by zannkukai — APPROVED** [review]

**Review by PascalRepond — APPROVED** [review]

File: projects/admin/src/app/api/ill-request-api.service.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  /**
+   * Constructor
+   * @param _recordService - RecordService
+   */
+  constructor(private _recordService: RecordService) {
+    super();
+  }
+
+  /**
+   * Get Ill requests by patron pid
```
Comment (jma) [inline projects/admin/src/app/api/ill-request-api.service.ts:41]: Can you add a sentence to tell that only the pending requests are considered.

File: projects/admin/src/app/circulation/circulation.scss [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ @import 'bootstrap/scss/variables';
+
+ .btn-hover {
+   color: $black;
+   background-color: $warning;
+   // could be better to use ``@extend .btn:hover`` but it doesn't work for hover property
+ }
+
+ .item {
+   margin-bottom: 0.25rem !important;
```
Comment (jma) [inline projects/admin/src/app/circulation/circulation.scss:28]: Can you use a bootstrap variable if exists?

File: projects/admin/src/app/circulation/patron/ill-request/ill-request-item/ill-request-item.component.html [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<div class="row item">
+  <div class="col-6" translate>
```
Comment (jma) [inline projects/admin/src/app/circulation/patron/ill-request/ill-request-item/ill-request-item.component.html:18]: Does it work with a button element as child?

File: projects/admin/src/app/circulation/patron/ill-request/ill-request.component.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+   * @param _patronService - PatronService
+   */
+  constructor(
+    private _illRequestApiService: IllRequestApiService,
+    private _patronService: PatronService
+  ) { }
+
+  /** onInit hook */
+  ngOnInit(): void {
+    this.illRequests$ = this._patronService.currentPatron$.pipe(
```
Comment (jma) [inline projects/admin/src/app/circulation/patron/ill-request/ill-request.component.ts:44]: does the current patron change after this component initialization?

File: projects/admin/src/app/circulation/patron/main/main.component.html [hunk 9]
```diff
@@ -66,6 +66,19 @@
...
+      <li class="nav-item">
+        <a
+          id="ill-tab"
+          class="nav-link"
+          routerLinkActive="active"
+          [routerLinkActiveOptions]="{exact: true}"
+          [routerLink]="['/circulation', 'patron', barcode, 'ill']">
+          {{ 'Interlibrary loan' | translate }}
+          <span *ngIf="getCirculationStatistics('ill') > 0 as stat" class="badge badge-info font-weight-normal">
+            {{ getCirculationStatistics('ill') }}
```
Comment (jma) [inline projects/admin/src/app/circulation/patron/main/main.component.html:78]: Should be `stat` to avoid multiple method calls.

**Review by jma — APPROVED** [review]

**Pull request merged**